src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2660,7 +2662,19 @@ MultiRouterPlannableQuery(Query *query, RelationRestrictionContext *restrictionC
 			char partitionMethod = PartitionMethod(distributedTableId);
 
 			if (!(partitionMethod == DISTRIBUTE_BY_HASH || partitionMethod ==
-				  DISTRIBUTE_BY_NONE))
+				  DISTRIBUTE_BY_NONE || partitionMethod == DISTRIBUTE_BY_RANGE))
+			{
+				return false;
+			}
+
+			hasHashDistributedTable |= partitionMethod == DISTRIBUTE_BY_HASH;
```

Comment:
> This kind of joins are handled by repartitioning.

Yes, what I meant was we already allow such combinations in the other planners/executors, why should we disallow it here?

> I do not see a value allowing some of them in router execution. T

I think the value here is that both planner/executor pairs behave similarly for the similar queries (in terms of partition types.)

> The same query could become magically supported and not supported between 2 setups (test and prod).

That's right. But, if you're dealing with range partitioned tables the responsibility of maintaining co-location is is always on you. If we think in that way, we shouldn't allow router planner for range partitioned tables at all.

So, I don't see it similar to the case that we've disallowed router planner hitting two shards on the same worker in [here](https://github.com/citusdata/citus/pull/1010/commits/d745d7bf70cbf12318b84d4971ddab22224f2c84) (i.e., `RelationPrunesToMultipleShards ()`)